addons/website_mail_channel/static/src/snippets/s_channel/options.js

```diff
@@ -8,12 +8,38 @@ var wUtils = require('website.utils');
 var _t = core._t;
 
 options.registry.Channel = options.Class.extend({
+    publicChannels: [],
+
+    /**
+     * @override
+     */
+    start() {
+        const prom = this._getPublicChannels().then(channels => {
+            this.publicChannels = channels;
+        });
+        return Promise.all([this._super(...arguments), prom]);
+    },
     /**
      * @override
      */
     cleanForSave: function () {
         this.$target.addClass('d-none');
     },
+    /**
+     * when a new snippet dropped from d&d menu :
```

Comment:
Should start by an uppercase, but remove that sentence entirely not need to repeat what the `onBuilt` method is for